### PR TITLE
Complete Day 3

### PR DESCRIPTION
**Pull request by jmcolella**

### CODE REVIEW

**Comment by maxnovak** [comment]
Please add a description to this PR :)

File: src/day_3/solution.rs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    if item_7 == tree_char {
+      let count = tree_tracker.entry("position_7").or_insert(0);
+      *count += 1;
+    }
+
+    if position_2_down_line == current_line {
+      let item_2_down = find_item(&slice_record, position_2_down);
+
+      if item_2_down == tree_char {
+        let count = tree_tracker.entry("position_2_down").or_insert(0);
```
Comment (kengorab) [inline src/day_3/solution.rs:96]: +1 for the `entry` api, pretty powerful

File: src/day_3/solution.rs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    }
+
+    let item_1 = find_item(&slice_record, position_1);
+    let item_3 = find_item(&slice_record, position_3);
+    let item_5 = find_item(&slice_record, position_5);
+    let item_7 = find_item(&slice_record, position_7);
+
+
+    if item_1 == tree_char {
+      let count = tree_tracker.entry("position_1").or_insert(0);
```
Comment (kengorab) [inline src/day_3/solution.rs:96]: +1 for the `entry` api, pretty powerful

**Review by kengorab — APPROVED** [review]
Looks ⭐  to me

File: src/main.rs [hunk 3]
```diff
@@ -1,6 +1,7 @@
-mod day_2;
+mod day_3;
 
 fn main() {
     // day_1::run()
-    day_2::solution::run()
+    // day_2::solution::run();
```
Comment (maxnovak) [inline src/main.rs:5]: -_-

File: src/day_3/solution.rs (hunk 2 above)
Comment (maxnovak) [reply src/day_3/solution.rs:96]: :O 
I know no rust, but just looked this up - very cool feature of the language

**Pull request merged**